packages/@expo/cli/e2e/__tests__/export-dom-test.ts

```diff
@@ -35,6 +35,11 @@ describe('Export DOM Components', () => {
 
   beforeAll(async () => {
     projectRoot = await setupTestProjectWithOptionsAsync('dom-export', 'with-dom');
+
+    // TODO(kudo,20250304): Remove this once we publish `@expo/metro-config` with DOM components fixes.
+    const srcMetroConfig = path.resolve(__dirname, '../../../metro-config/build');
```

Comment:
I think we have something for this already https://github.com/expo/expo/blob/784214b7a200b2492e1dce40141276acf80e856e/packages/%40expo/cli/e2e/__tests__/export-monorepo-test.ts#L19-L23

Reply:
that one is neat.

try it out but it doesn't work well for `@expo/metro-config`, because the dependency chain. i guess we need `resolutions` further to override all deps to use the linked version. i can add resolutions to the linkExpoPackages in a following up pr later if you feel that's good to add.

```sh
$ find . -name metro-config
./node_modules/expo/node_modules/@expo/metro-config
./node_modules/@expo/cli/node_modules/@expo/metro-config
./node_modules/@expo/metro-config
```

Reply:
Damn, we should fix that by adding the `resolutions` + `override` to this linking logic here: https://github.com/expo/expo/blob/c00dcf72168eeab82376f450d0cd08db156c88a7/packages/%40expo/cli/e2e/__tests__/utils.ts#L115-L129

Reply:
Created a PR to fix both the outdated hashes in assertions, and to force-overwrite the package through `resolutions` and `overrides`: https://github.com/expo/expo/pull/35363